pkg/proc/target.go

```diff
@@ -108,6 +108,13 @@ func (sr StopReason) String() string {
 	}
 }
 
+func (t *Target) StopReason() StopReason {
```

Comment:
I don't think that a method meant to inspect the reason why the target stopped should retroactively change it based on things that happened after the target stopped.

Reply:
I was thinking about it as "even if it hadn't stopped for t.stopReason, it would have stopped because a manual halt was issued".  The rest of the state is still there and we can surface the breakpoints, etc.

I agree, it does seem a little weird though. I will remove this and have DAP server check this separately from checking the StopReason.